contexts/SummitData.js

```diff
@@ -9,12 +16,31 @@ import queryEfd from "@/lib/api/efd";
 
 export const SummitDataContext = createContext({});
 
+const loadingReducer = (state, action) => {
+  const { type, dataSet } = action;
+
+  switch (type) {
+    case "complete":
+      return { ...state, [dataSet]: false };
+    case "loading":
+      return { ...state, [dataSet]: true };
+    default:
+      throw new Error(`Unhandled action type: ${type}`);
+  }
+};
+
+const defaultLoadState = {
+  currentData: true,
+  hourlyData: true,
+  dailyData: true,
+};
+
 export const SummitDataProvider = ({ children }) => {
   const [currentData, setCurrentData] = useState();
   const [hourlyData, setHourlyData] = useState();
   const [dailyData, setDailyData] = useState();
   const [error, setError] = useState(false);
-  const [loading, setLoading] = useState(true);
+  const [loading, dispatch] = useReducer(loadingReducer, defaultLoadState);
```

Comment:
tracking the individual loading states of the datasets, trying to make them appear at the right times and not access data when it's undefined was too tricky with a single global loading state